### PR TITLE
feat: add SimulationDetails component

### PR DESCRIPTION
**Pull request by dbrans**
## **Description**
Adds the SimulationDetails component from extension without fiat.

Note: 
Patch changes implementing the support of simulations to `TransactionController` which has no effect with the current flows.
Please see the `TransactionController` patch is updated on this PR https://github.com/MetaMask/core/pull/4303 

DONE:
- [x] useAsyncResult
- [x] types.ts
- [x] useBalanceChanges
- [x] sortBalanceChanges
- [x] formatAmount
- [x] AmountPill
- [x] AssetPill (once https://github.com/MetaMask/metamask-mobile/pull/9379 has been merged)
- [x] BalanceChangeRow
- [x] BalanceChangeList
- [x] SimulationDetails
- [x] SimulationDetails.stories

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1716

## **Manual testing steps**

N/A 

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/7644512/b6ee063c-2473-4c3e-b1ad-afba083a11f3

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/AnimatedSpinner/index.js [hunk 1]
```diff
@@ -1,24 +1,67 @@
...
+    static: {
+      borderRadius: 64,
+      width: 36,
+      height: 36,
+      iconSize: 36,
+    },
+  },
+};
+
+const createStyles = (colors, measures) =>
```
Comment (OGPoyraz) [inline app/components/UI/AnimatedSpinner/index.js:48]: Both sizes could be seen here, no visual differences, just a smaller one is added. 

https://github.com/MetaMask/metamask-mobile/assets/7644512/6df13905-412e-4ba3-a1d7-1218c06e5af3

File: app/components/UI/SimulationDetails/AmountPill.styles.ts
Comment (matthewwalsh0) [inline app/components/UI/SimulationDetails/AmountPill.styles.ts:1]: As there's now three files for each component, is it worth creating sub-directories inside `SimulationDetails`?

File: app/components/UI/SimulationDetails/types.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+   * (increase) or negative (decrease).
+   *
+   * Example: If an asset supports 18 decimals, an `amount` of 1.5 represents
+   * 1.5 tokens, or more precisely, 1.5 * 10^18 of the smallest divisible unit.
+   */
+  amount: BigNumber;
+}>;
+
+/** Token standards supported by simulation. */
+export enum SimulationTokenStandard {
```
Comment (matthewwalsh0) [inline app/components/UI/SimulationDetails/types.ts:67]: I'm assuming if the patch gets merged first, we can remove all these as we want to use the `TransactionController` types?

File: locales/languages/en.json [hunk 3]
```diff
@@ -3008,5 +3008,15 @@
         "turn_off": "Turn off"
       }
     }
+  },
+  "simulation_details": {
+    "reverted": "This transaction is likely to fail",
```
Comment (matthewwalsh0) [inline locales/languages/en.json:3013]: Should these be ordered alphabetically?

File: app/components/UI/SimulationDetails/AmountPill.styles.ts
Comment (OGPoyraz) [reply app/components/UI/SimulationDetails/AmountPill.styles.ts:1]: Done in https://github.com/MetaMask/metamask-mobile/pull/9410/commits/9440e0a1f5b8313c2c03d37d462ecc49d1690fdb

File: app/components/UI/SimulationDetails/types.ts (hunk 2 above)
Comment (OGPoyraz) [reply app/components/UI/SimulationDetails/types.ts:67]: Yes I will keep this on hold, and probably update patch in this PR.

File: locales/languages/en.json (hunk 3 above)
Comment (OGPoyraz) [reply locales/languages/en.json]: Sorted in https://github.com/MetaMask/metamask-mobile/pull/9410/commits/204f0cdd1552d8e6dcced8f1ba3609fec836acb2

**Review by OGPoyraz — DISMISSED** [review]

File: app/components/UI/SimulationDetails/types.ts (hunk 2 above)
Comment (OGPoyraz) [reply app/components/UI/SimulationDetails/types.ts]: This is fixed as we updated the patch.

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by OGPoyraz — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by tommasini — APPROVED** [review]
LGTM! Maybe worth considering convert AnimatedSpinner component to typescript and functional component!

Also if possible add that the changes on this patch were added on the core branch of transaction controller mobile patch: `patch/mobile-transaction-controller-13-0-0`

Awesome work!

**Comment by OGPoyraz** [comment]
> LGTM! Maybe worth considering convert AnimatedSpinner component to typescript and functional component!
> 
> Also if possible add that the changes on this patch were added on the core branch of transaction controller mobile patch: `patch/mobile-transaction-controller-13-0-0`
> 
> Awesome work!

Patch information added to the description, thanks for raising this

**Pull request merged**